sdk/keyvault/azure-keyvault-administration/tests/_async_test_case.py

```diff
@@ -0,0 +1,102 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+import os
+
+import pytest
+from azure.keyvault.administration import ApiVersion
+from azure.keyvault.administration._internal.client_base import DEFAULT_VERSION
+from devtools_testutils import AzureRecordedTestCase
+
+
+class BaseClientPreparer(AzureRecordedTestCase):
+    def __init__(self, **kwargs) -> None:
+        hsm_playback_url = "https://managedhsmvaultname.vault.azure.net"
+        container_playback_uri = "https://storagename.blob.core.windows.net/container"
+        playback_sas_token = "fake-sas"
+
+        if self.is_live:
+            self.managed_hsm_url = os.environ.get("AZURE_MANAGEDHSM_URL")
+            storage_name = os.environ.get("BLOB_STORAGE_ACCOUNT_NAME")
+            storage_endpoint_suffix = os.environ.get("KEYVAULT_STORAGE_ENDPOINT_SUFFIX")
+            container_name = os.environ.get("BLOB_CONTAINER_NAME")
+            self.container_uri = "https://{}.blob.{}/{}".format(storage_name, storage_endpoint_suffix, container_name)
+
+            self.sas_token = os.environ.get("BLOB_STORAGE_SAS_TOKEN")
```

Comment:
Looks like we'll need an `add_general_regex_sanitizer` call to replace `self.sas_token` with `playback_sas_token` in recordings